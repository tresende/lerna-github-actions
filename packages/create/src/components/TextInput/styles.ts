import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin: 1rem 0;
`

export const Label = styled.label`
  font-size: 1.4rem;
`

export const Input = styled.input`
  border: 0;
  padding: 0.6rem 0;
  margin-top: 0.5rem;
  border-bottom: 1px solid lightgray;

  &:focus,
  &:focus {
    outline: none;
  }
`
