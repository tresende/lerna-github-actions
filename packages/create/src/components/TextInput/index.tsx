import React, { ChangeEvent } from 'react'

import * as S from './styles'

type TextInputProps = {
  label: string
  value?: string | Date
  type?: 'text' | 'date'
  onChange: (value: string) => void
}

const TextInput = ({ label, value, type = 'text', onChange }: TextInputProps) => {
  const name = label.toLocaleLowerCase().replaceAll(' ', '_')

  const handleChange = ({ currentTarget }: ChangeEvent<HTMLInputElement>) => {
    onChange(currentTarget.value)
  }

  return (
    <S.Wrapper>
      <S.Label htmlFor={name}>{label}</S.Label>
      <S.Input value={(value ?? '') as string} placeholder={label} type={type} onChange={handleChange} name={name} />
    </S.Wrapper>
  )
}

export default TextInput
