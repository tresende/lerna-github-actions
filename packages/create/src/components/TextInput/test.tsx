import React from 'react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import { render, screen } from '@testing-library/react/pure'

import TextInput from '.'

const props = {
  text: 'FAKE',
  value: 'FAKE TEXT',
  onChange: jest.fn()
}

describe('<TextInput />', () => {
  beforeAll(() => {
    render(<TextInput {...props} />)
  })

  it('Should render input text by default', () => {
    expect(screen.getByRole('textbox')).toHaveAttribute('type', 'text')
  })

  it('Should call onChange on type text input', () => {
    const text = 'FAKE'
    const input = screen.getByRole('textbox')
    userEvent.type(input, text)

    expect(props.onChange).toHaveBeenCalledTimes(text.length)
  })

  it('Should call render with label', () => {
    expect(screen.getByText(props.text)).toBeInTheDocument()
    expect(screen).toMatchSnapshot()
  })
})

describe('<TextInput type="date">', () => {
  it('Should render date input', () => {
    render(<TextInput {...props} type="date" text="date" />)

    expect(screen.getByPlaceholderText(/date/i)).toHaveAttribute('type', 'date')
  })
})
