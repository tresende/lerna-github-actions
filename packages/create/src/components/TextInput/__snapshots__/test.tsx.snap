// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> Should call redenr with label 1`] = `
<div>
  <div
    class="sc-bdvvtL iXMHdx"
  >
    <label
      class="sc-gsDKAQ iYacNX"
      for="FAKE"
    >
      FAKE
    </label>
    <input
      class="sc-dkPtRN iQffaK"
      name="FAKE"
      placeholder="FAKE"
      type="text"
      value=""
    />
  </div>
</div>
`;
